### PR TITLE
[Automation] Bootstrap cluster using cephadm-ansible wrapper modules,…

### PR DESCRIPTION
**Pull request by pranavprakash20**
… get and set configs

Signed-off-by: Pranav <prprakas@redhat.com>

# Description
Tests included
CEPH-83575205 Bootstrap a cluster using cephadm-ansible module 'cephadm_bootstrap' with overriding existing ceph keys
CEPH-83575215 Get 'mgr' config using cephadm-ansible module 'ceph_config'
CEPH-83575214 Set 'osd' config using cephadm-ansible module 'ceph_config'

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by pranavprakash20** [comment]
run log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-UR4TGG/

File: playbooks/bootstrap-cluster-overide-exisitng-keys.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  hosts: admin
+  gather_facts: false
+  become: true
+  any_errors_fatal: true
+
+  tasks:
+    - name: Bootstrap with dashboard set to `true` without credenatials
+      cephadm_bootstrap:
+        mon_ip: "{{ mon_ip }}"
+        allow_overwrite: true
```
Comment (vamahaja) [inline playbooks/bootstrap-cluster-overide-exisitng-keys.yml:13]: Missing `dashboard` key from module ?

File: suites/pacific/cephadm/test_cephadm_bootstrap_with_overriding_exisiting_ceph_keys.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+# CEPH-83575215
+# CEPH-83575214
+#===============================================================================================
+tests:
+  - test:
+      name: setup install pre-requisistes
+      desc: Setup phase to deploy the required pre-requisites for running the tests.
+      module: install_prereq.py
+      abort-on-fail: true
+
```
Comment (vamahaja) [inline suites/pacific/cephadm/test_cephadm_bootstrap_with_overriding_exisiting_ceph_keys.yaml:27]: Don't we need to bootstrap cluster first ?

File: tests/cephadm/test_cephadm_ansible_wrapper_set_osd_config.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+log = Log(__name__)
+
+CEPHADM_ANSIBLE_PATH = "/usr/share/cephadm-ansible"
+
+
+class ConfigNotFoundError(Exception):
+    pass
+
+
+def put_cephadm_ansible_playbook(node, playbook):
```
Comment (vamahaja) [inline tests/cephadm/test_cephadm_ansible_wrapper_set_osd_config.py:17]: better to have comman place for this function

**Review by vamahaja — CHANGES_REQUESTED** [review]
IMO we should combine all these TCs in single test module instead of having separate module.

File: playbooks/bootstrap-cluster-overide-exisitng-keys.yml (hunk 1 above)
Comment (pranavprakash20) [reply playbooks/bootstrap-cluster-overide-exisitng-keys.yml:14]: Added

File: suites/pacific/cephadm/test_cephadm_bootstrap_with_overriding_exisiting_ceph_keys.yaml (hunk 2 above)
Comment (pranavprakash20) [reply suites/pacific/cephadm/test_cephadm_bootstrap_with_overriding_exisiting_ceph_keys.yaml:27]: Bootstrap is done in the next step

**Comment by pranavprakash20** [comment]
@vamahaja As suggested, clubbed the tc's together and the bootstrap tc which was existing was reused

**Comment by pranavprakash20** [comment]
New run log :  http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-K9D9VM/

File: tests/cephadm/test_cephadm_ansible_wrapper_set_osd_config.py (hunk 3 above)
Comment (pranavprakash20) [reply tests/cephadm/test_cephadm_ansible_wrapper_set_osd_config.py]: Added to cephci/cli/cephadm/ansible.py

File: cli/cephadm/ansible.py [hunk 4]
```diff
@@ -43,3 +48,15 @@ def run_playbook(
             raise CephAdmAnsibleError(
                 f"Failed to execute cephadm ansible playbook '{playbook}'"
             )
+
+    def put_cephadm_ansible_playbook(
```
Comment (vamahaja) [inline cli/cephadm/ansible.py:52]: Move this method to `cli/utils.py`

**Review by vamahaja — CHANGES_REQUESTED** [review]

**Comment by pranavprakash20** [comment]
New run log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-G3VG3Y/

**Review by MohitBis — DISMISSED** [review]
LGTM

**Review by vamahaja — CHANGES_REQUESTED** [review]

**Review by vamahaja — DISMISSED** [review]

**Comment by pranavprakash20** [comment]
New run log :  http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-T7WVPU/

**Review by vamahaja — APPROVED** [review]

**Comment by vdas-redhat** [comment]
LGTM

**Review by vdas-redhat — APPROVED** [review]

**Pull request merged**